presto-main/src/main/java/com/facebook/presto/operator/TaskContext.java

```diff
@@ -202,77 +226,36 @@ public void addCompletedDriverGroup(Lifespan driverGroup)
         return pipelineContexts;
     }
 
-    public synchronized ListenableFuture<?> reserveMemory(long bytes)
-    {
-        checkArgument(bytes >= 0, "bytes is negative");
-
-        ListenableFuture<?> future = queryContext.reserveMemory(bytes);
-        memoryReservation.getAndAdd(bytes);
-        return future;
-    }
-
-    public synchronized ListenableFuture<?> reserveRevocableMemory(long bytes)
-    {
-        checkArgument(bytes >= 0, "bytes is negative");
-
-        ListenableFuture<?> future = queryContext.reserveRevocableMemory(bytes);
-        revocableMemoryReservation.getAndAdd(bytes);
-        return future;
-    }
-
-    public synchronized ListenableFuture<?> reserveSystemMemory(long bytes)
-    {
-        checkArgument(bytes >= 0, "bytes is negative");
-        ListenableFuture<?> future = queryContext.reserveSystemMemory(bytes);
-        systemMemoryReservation.getAndAdd(bytes);
-        return future;
-    }
-
     public synchronized ListenableFuture<?> reserveSpill(long bytes)
     {
         checkArgument(bytes >= 0, "bytes is negative");
         return queryContext.reserveSpill(bytes);
     }
 
-    public synchronized boolean tryReserveMemory(long bytes)
+    public synchronized void freeMemory(long bytes)
```

Comment:
Call this `freeUserMemory`. Assert that `bytes` is non-negative. Delegate to `queryContext.updateUserMemory`.